### ISSUE TITLE
Could this run on the same server as the gateway?

### ISSUE THREAD

**Issue by flatsiedatsie** [hidaris/thingtalk#6]
This library looks really nice. Would it theoretically be possible to run it on the same server as the Gateway? Perhaps with a different port? Or would it then no longer be auto-discovered?

**Comment by hidaris** [comment]
@flatsiedatsie Hello, theoretically it should be supported, in the earliest time of writing this project, my first application is zigbee2mqtt access to webthings, when the test was done on gateway, but since then, I have not used gateway again, changed to my own thing ui, from the point of view of self-description, there should not be incompatible changes, at present still can pass the webthings api test, in any case, I will test the following again at the weekend.

**Comment by flatsiedatsie** [comment]
Cool, thanks. 

You may also enjoy the Zigbee2MQTT addon that the Webthings Gateway now has.

**Comment by hidaris** [comment]
@flatsiedatsie Hi, I tried it over the weekend, unfortunately the current thingtalk websocket implementation is not fully compatible with webthings, changing from multiple endpoints to a single endpoint, the reason for this is because when trying to use tools like ngrok or frp for intranet penetration, the webthings design will require a high number of connections to the proxy server, if you are interested in this project, I can provide you with a webthings compatible model.

**Comment by hidaris** [comment]
> Cool, thanks.
> 
> You may also enjoy the Zigbee2MQTT addon that the Webthings Gateway now has.

Yes, I went to look at this new plugin and it's great for gateway users
